Credential configuration attributes (#60)
hashicorp/terraform-provider-kubernetes-alpha: docs/usage.md

## Provider configuration

> The provider accepts the following configuration attributes under the `provider` block.

Reviewer: I like the type description, but I think we should still denote whether something is required or optional.  Aligned with https://www.terraform.io/docs/providers/kubernetes/index.html#argument-reference

Author: Good point. Everything is optional so I'll just add that as a note before the list. I feel like adding the word "optional" to all the entries just clutters it up? WDYT?

Author: Added a line noting everything is optional but a meaningful set of options should be configured.

* `config_context_user` - (string) (env-var: `KUBE_CTX_USER`) User entry to associate to the current context (from kubeconfig).
* `config_context_cluster` - (string) (env-var: `KUBE_CTX_CLUSTER`) Cluster entry to associate to the current context (from kubeconfig).
* `token` - (string) (env-var: `KUBE_TOKEN`) Token is a bearer token used by the client for request authentication.
* `insecure` - (boolean) (env-var: `KUBE_INSECURE`) Disregard invalid TLS certificates _(default false)_.
* `exec` - (object) Exec-based authentication plugin.
  * `api_version` - (string) Version of the "client.authentication.k8s.io" API which the plugin implements.
  * `command` - (string) The plugin executable (absolute path, or expects the plugin to be in OS PATH).
  * `env` - (map string to string) Environment values to set on the plugin process.
  * `args` - (list of strings) Command line arguments to the plugin command.

Due to the internal design of this provider, access to a responsive API server is required both during PLAN and APPLY. The provider makes calls to the Kubernetes API to retrieve metadata and type information during all stages of Terraform operations.

### Credentials

> For authentication, the provider can be configured with identity credentials sourced from either a `kubeconfig` file, explicit values in the `provider` block or a combination of both.

Reviewer: ```suggestion
For authentication, the provider can be configured with identity credentials sourced from either a `kubeconfig` file, explicit values in the `provider` block, or a combination of both.
```

Added an Oxford comma, turns out I do care about them, despite what Vampire Weekend may say.

> Setting the `config_path` attribute to the path of a `kubeconfig` file will drive the provider to load it and use the credential values in it. When `config_path` is not set **NO EXTERNAL KUBECONFIG WILL BE LOADED**. Specifically, $KUBECONFIG environment variable is **NOT** considered.

Reviewer: ```suggestion
If the `config_path` attribute is set to the path of a `kubeconfig` file, the provider will load it and use the credential values in it. When `config_path` is not set **NO EXTERNAL KUBECONFIG WILL BE LOADED**. Specifically, $KUBECONFIG environment variable is **NOT** considered.
```

Take note of the `current-context` configured in the file. You can override it using the `config_context` provider attribute.

> When combined, the values loaded from the `kubeconfig` file are overridden by the corresponding values configured on `provider` block attributes.

Reviewer: ```suggestion
If both `kubeconfig` and static credentials are defined in the `provider` block, the provider will prefer the static credentials and ignore the `kubeconfig`.
```

> Generaly, there are four major options for providing identity information to the provider for authentication purposes:

Reviewer: ```suggestion 
There are five options for providing identity information to the provider for authentication purposes:
```

> 

Reviewer: ```suggestion

* a kubeconfig
```
